plugins/mongodb/README.md

```diff
@@ -0,0 +1,202 @@
+
+### Inserting Documents
+
+The plugin supports inserting documents one at a time using `insert_one` and inserting multiple documents at once using insert_many. To insert a document you just need to call insert() inside the embedded script and the plugin will call `insert_one` if you pass in a single record or scalar values. If you pass in a dataset the plugin will automatically call `insert_many` for you.
+
+```
+dataset({INTEGER inserted_count}) insertMany(dataset(layoutEmployee) employees) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    insert({$employees});
+ENDEMBED;
+insertMany(ds);
+
+bool insertOne(STRING first, STRING last, REAL salary) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    insert({first: $first, last: $last, salary: $salary}); 
+ENDEMBED;
+insertOne();
+```
+
+The first example inserts a dataset with the layoutEmployee record and the second example creates a bson document and inserts some scalar values into it then adds it to the collection. 
+
+### Finding Documents
+
+The plugin allows the ECL user to choose between find_one and find_many. They both take a single document as the filter for the find operation.
+
+```
+dataset({STRING first, STRING last, REAL salary}) findOne() := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    find_one({first: "John", last: "Doe"});
+ENDEMBED;
+findOne();
+
+dataset({STRING first, STRING last, REAL salary}) findMany() := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    find({first: "John", last: "Doe"});
+ENDEMBED;
+findMany();
+```
+
+There is support for providing a projection to the find function. If no projection is given then MongoDB returns all fields from the document, and the ECL engine must look through every field trying to find the ones declared in the return type by the ECL user.
+
+```
+dataset({STRING first, STRING last, REAL salary}) findMany(INTEGER include) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    find({first: "John", last: "Doe"}, {first: $include, last: $include, salary: $include);
+ENDEMBED;
+findMany(1);
+```
+
+### Updating Documents
+
+The plugin can update one or many documents which is specified by the user. it returns a single Record with two fields holding the number of documents that were matched by the filter and another holding the number of documents that were modified by the operation. The update methods can take either two documents or a document and a pipeline. A pipeline follows the structure [{doc}, ... ].
+
+```
+dataset({Integer matched_count, Integer modified_count}) updateOne(REAL raise) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
```

Comment:
Is the RECORD definition returned by MongoDB update functions standardized?  If so, it may make sense to create an exported definition within your .ecllib file so that ECL users can simply use it rather than redefine it themselves every time (and get it wrong some of the time).